antlr/src/main/antlr/FSH.g4

```diff
@@ -1,12 +1,13 @@
 grammar FSH;
 
 doc:                entity*;
-entity:             alias | profile | extension | invariant;
+entity:             alias | profile | extension | invariant | instance;
 
 alias:              KW_ALIAS SEQUENCE EQUAL SEQUENCE;
 
 profile:            KW_PROFILE SEQUENCE sdMetadata+ sdRule*;
 extension:          KW_EXTENSION SEQUENCE sdMetadata* sdRule*;
+instance:           KW_INSTANCE SEQUENCE KW_INSTANCEOF SEQUENCE title? fixedValueRule*;
```

Comment:
This requires the exact order:
```
Instance: xxx
InstanceOf: yyy
Title: zzz
```
In our other types (profile and extension), the metadata can come in any order (via the `sdMetadata*`). For user-friendliness, I think we should do the same here.  If we make an `instanceMetadata` parser rule, this also gives us a good place to put further metadata if we come up with any.  The downside is that this means that we can't mark `InstanceOf` as _required_ in the grammar -- so we'll have to ensure it is specified when we parse it instead.

Reply:
I have mixed feelings about this one. As you point out, it's more user-friendly to not enforce a specific order to fields. The documentation implies, but does not outright state, that these should always be the first two keywords when defining an instance: 
> Instance specifications are introduced using the `Instance` and `InstanceOf` keywords.

And, all examples have `Instance` as the first and `InstanceOf` as the second. Enforcing the order like this is not something done by the grammar for other entities. And, thinking ahead to a potential future where there are multiple required metadata fields, it would be unusual to enforce an order. Therefore, I'll go ahead and make it an optional field from the grammar's perspective, and enforce its presence in `FSHImporter`.

Reply:
I'm comfortable with the ultimate decision here to make it an optional field in the grammar and enforce it in the importer. I have a similar feeling to @mint-thompson about how, in practice, it seems like it will always be in this exact order. But I will similarly concede that the benefits of flexibility outweigh the negatives.